src/routes/auth/index.ts

```diff
@@ -0,0 +1,147 @@
+			to: parsedBody.email,
+			from: context.env.SENDER_EMAIL,
+			subject: '[TorontoJS] Confirm your account',
+			html: emailHtmlTemplate
+		};
+		await sgMail.send(msg);
+
+		return context.json<StatusResponse>({ message: 'Sent an email successfully' }, StatusCodes.OKAY);
+	} catch (error) {
+		console.error('Error sending email', error);
+		return context.json<StatusResponse>({ message: 'Error sending email' }, StatusCodes.INTERNAL_SERVER_ERROR);
+	}
+});
+
+authRoutes.get('/activate', async (context: Context<EnvironmentBindings>) => {
+	try {
+		const token = context.req.query('token');
+		if (!token) {
+			return context.json({ message: 'Invalid or missing token' }, StatusCodes.BAD_REQUEST);
+		}
+
+		const email = await context.env.ACTIVATION_TOKENS.get(token);
+		if (!email) {
+			return context.json({ message: 'Invalid or expired token' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		const emailExists = await checkEmail(context.env.database, email);
+		if (!emailExists) {
+			return context.json({ message: 'User not found' }, StatusCodes.NOT_FOUND);
+		}
+
+		await activateProfile(context.env.database, email);
+
+		// Remove token after successful activation
+		await context.env.ACTIVATION_TOKENS.delete(token);
+
+		return context.json({ message: 'Account activated successfully' }, StatusCodes.OKAY);
+	} catch (error) {
+		return context.json<StatusResponse>({ message: 'Error activating an account' }, StatusCodes.INTERNAL_SERVER_ERROR);
+	}
+});
+
+authRoutes.post('/sign-in', async (context: Context<EnvironmentBindings>) => {
+	try {
+		const body = await context.req.json();
+		const parsedBody = SignInSchema.parse(body);
+
+		const storedPassword = await getPassword(context.env.database, parsedBody);
+
+		if (!storedPassword) {
+			return context.json<StatusResponse>({ message: 'Unauthorized requests' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		// Extract salt and hashed password from stored format
+		const [saltBase64, storedHashedPassword] = storedPassword.split(':');
+		if (saltBase64 === undefined) {
+			throw new Error('Salt not found');
+		}
+		const salt = new Uint8Array([...atob(saltBase64)].map((c) => c.charCodeAt(0)));
+
+		// Hash input password with the same salt
+		const inputHashedPassword = await hashPasswordPBKDF2(parsedBody.password, salt);
+
+		// Verify if the input password matches the stored password
+		if (inputHashedPassword !== storedHashedPassword) {
+			return context.json<StatusResponse>({ message: 'Unauthorized' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		// Generate session token
+		const sessionToken = crypto.randomUUID();
+		const hoursAhead = 1;
+		const tokenExpiry = String(Date.now() + hoursAhead * 60 * 60 * 1000);
+		const expiryAndUserEmail = `${tokenExpiry} ${parsedBody.email}`;
+		await context.env.SESSION_TOKENS.put(sessionToken, expiryAndUserEmail);
+
+		context.header('Set-Cookie', `auth_token=${sessionToken}; HttpOnly; Secure; SameSite=Strict; Expires=${tokenExpiry}; Path=/; Domain=torontojs.com`);
```

Comment:
Probably wouldn't put a domain in if you want people to test locally

Reply:
I think the `NODE_ENV` variable is set, we can use that to determine and only add a domain if it is "production"

Reply:
Unless you have a reason to, you can just not set a cookie domain. By default it will be the origin.